sdk/core/azure-core/src/http/log_policy.cpp

```diff
@@ -37,13 +37,32 @@ inline void AppendHeaders(
   }
 }
 
+inline std::string GetUrlWithoutQuery(Url url)
+{
+  std::string urlString;
+  if (!url.GetScheme().empty())
+  {
+    urlString += url.GetScheme() + "://";
+  }
+  urlString += url.GetHost();
+  if (url.GetPort() != 0)
+  {
+    urlString += ":" + std::to_string(url.GetPort());
+  }
+  if (!url.GetPath().empty())
+  {
+    urlString += "/" + url.GetPath();
+  }
+  return urlString;
+}
+
 inline std::string GetRequestLogMessage(LogOptions const& options, Request const& request)
 {
   auto const& requestUrl = request.GetUrl();
 
   std::ostringstream log;
   log << "HTTP Request : " << request.GetMethod().ToString() << " "
-      << requestUrl.GetUrlWithoutQuery();
+      << GetUrlWithoutQuery(requestUrl);
```

Comment:
Yep - better accept `ostringstream& log` as a parameter for `GetUrlWithoutQuery()`.

Reply:
What's the benefit? Right now the log ostringstream change is all in one place, I don't know if it's a good idea to have that be passed to the `GetUrlWithoutQuery()` method.